app/furniture/marketplace/cart.rb

```diff
@@ -26,7 +26,11 @@ def product_total
       end
     end
 
-    delegate :delivery_fee, to: :marketplace
+    def delivery_fee
+      return marketplace.delivery_fee if delivery_address.present?
```

Comment:
I assume this logic is covering the fact that the `cart.delivery_fee` cannot be calculated without a `delivery_address` so it sets `delivery_fee` to zero.

Reply:
Exactly.